pkgs/development/python-modules/bertopic/default.nix

```
@@ -0,0 +1,93 @@
+{
+  buildPythonPackage,
+  fetchPypi,
+  hdbscan,
+  lib,
+  llvmlite,
+  numpy,
+  pandas,
+  plotly,
+  pytestCheckHook,
+  scikit-learn,
+  sentence-transformers,
+  setuptools,
+  tqdm,
+  umap-learn,
+
+  # extras: datamap
+  datamapplot,
+  matplotlib,
+
+  # extras: fastembed
+  fastembed,
+
+  # extras: gensim
+  gensim,
+
+  # extras: spacy
+  spacy,
+
+  # extras: vision
+  accelerate,
+  pillow,
+}:
+
+buildPythonPackage rec {
+  pname = "bertopic";
+  version = "0.17.3";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-P4dtLq3yA333nwWbc0k11+0utvUwvMbHoZ4aj10suro=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    hdbscan
+    umap-learn
+    numpy
+    pandas
+    plotly
+    scikit-learn
+    sentence-transformers
+    tqdm
+    llvmlite
+  ];
+
+  optional-dependencies = {
+    datamap = [
+      datamapplot
+      matplotlib
+    ];
+    fastembed = [ fastembed ];
+    gensim = [ gensim ];
+    spacy = [ spacy ];
+    vision = [
+      accelerate
+      pillow
+    ];
+  };
+
+  # In order to make pythonImportsCheck work, NUMBA_CACHE_DIR env var need to
+  # be set to a writable dir (https://github.com/numba/numba/issues/4032#issuecomment-488102702).
+  # pythonImportsCheck has no pre* hook, use checkPhase to workaround that.
+  checkPhase = ''
+    export NUMBA_CACHE_DIR="$(mktemp -d)"
+  '';
+
+  pythonImportsCheck = [ "bertopic" ];
+
+  nativeCheckInputs = [
```

Actually no, there's a better way:
```suggestion
  pythonImportsCheck = [ "bertopic" ];

  nativeCheckInputs = [
    writableTmpDirAsHomeHook
```
setting writable `HOME` is also enough

Also, some tests fail, skip them (and write a comment with an explanation too)